File: core/scripts/Java-Library/src/chh/math/geo/Rhombus.java
package chh.math.geo;

/**
 * A rhombus is a quadrilateral with four congruent sides.
 * Theorem: a rhombus is a parallelogram.
 * @author Christopher H. Harvey <chrisharvey2pi@gmail.com>
 * @version 2014.03.27
 */
public class Rhombus extends Parallelogram {
	/** 
	 * Creates a new Rhombus object.
	 * @param a the side of this rhombus
	 * @param ang the angle, in degrees, between sides a and b
	 */
	public Rhombus(Length a, double ang) {
		super(a, a, ang);
	}	
}

File: core/scripts/geo/Rhombus.js
var Util = require('../Util.js');
var Parallelogram = require('./Parallelogram.js');

/**
  * A rhombus is a quadrilateral with four congruent sides.
  * Theorem: a rhombus is a parallelogram.
  * @extends     Parallelogram
  * @param `a`   a Length object: the base of this rhombus
  * @param `ang` an Angle object: in degrees, between the base and the left side
  */
function Rhombus(a, ang) {
  Parallelogram.call(this, a, a, ang);
}
Util.extend(Rhombus, Parallelogram);

module.exports = Rhombus;
